src/app/clusters/software-diagnostics-server/software-diagnostics-cluster.h

```diff
@@ -84,17 +84,26 @@ class SoftwareDiagnosticsServerCluster : public DefaultServerCluster,
         {
         case SoftwareDiagnostics::Attributes::CurrentHeapFree::Id: {
             uint64_t value;
-            ReturnErrorOnFailure(LOGIC::GetCurrentHeapFree(value));
+            if (LOGIC::GetCurrentHeapFree(value) != CHIP_NO_ERROR)
```

Comment:
It seems really strange to cover up all errors here.  Why are we doing that?

Reply:
Because this is what I inherited: https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/software-diagnostics-server/software-diagnostics-server.cpp#L112

If I do any sort of "be better" I will pay flash cost of logging. I agree that this is dodgy...

Reply:
Right, the old thing did not cover up all errors.